### ISSUE TITLE
Flax unresponsive while doing work

### ISSUE THREAD

**Issue by slipster216** [FlaxEngine/FlaxEngine#1467]
**Issue description:**
Flax often goes unresponsive while it's doing something. 

**Steps to reproduce:**
Open a project
Select "Generate Scripts Project Files" from the menu
Select file again, and notice the menu does not come up. 

Expect:
- Progress bar, or other action showing that flax will be unresponsive until the operation is finished.

**Flax version:**
1.6

**Comment by Menotdan** [comment]
![image](https://github.com/FlaxEngine/FlaxEngine/assets/32620310/566324f9-7705-4b30-8c06-fae42e4206ee)
Is this the kind of progress bar you mean? Its not very obvious that it will halt the editor being responsive there, but it exists. However, my context menu still opens when the editor is busy.. maybe that's a separate issue.

**Comment by slipster216** [comment]
Actually I have this issue completely wrong- I can get this to happen even without invoking any kind of action that triggers the progress bar, but that just makes it easier. Basically, once I invoke the action it can take anywhere from 1-10 clicks before the file menu bar works again. I've posted a video here:

https://youtu.be/eOCYMP8cneM

Basically the first time I do it everything works fine. The second time takes 5 clicks - you can't easily see them but you can see the window's icons changing as I click each time. This made me think the command was making me lose focus on the window, which might be true because (I think) this is just lauching the .command file? But then the next time it took like 10 clicks before it responded, though the window icons didn't change. 

This could be an osx thing- either way it's low priority, for sure, but does make the program feel a little cheap.

**Comment by Menotdan** [comment]
Yeah it looks like the window gets weirdly unfocused for seemingly no reason.
